core/js/typespecs/api_types.ts

```diff
@@ -170,6 +171,19 @@ function generateBaseEventOpSchema(objectType: ObjectTypeWithEvent) {
       .describe(
         "Human-identifying attributes of the span, such as name, type, etc.",
       ),
+    origin: z
+      .object({
+        object_type: eventObjectType.describe(
+          "Type of the object the event is originating from.",
+        ),
+        object_id: z
+          .string()
+          .uuid()
+          .describe("ID of the object the event is originating from."),
+        id: z.string().describe("ID of the original event."),
```

Comment:
I wonder if for posterity we should include an `_xact_id`, in case the row changes or is deleted. We don't have to use it in the UI right now, but wondering if it's worth including.

Reply:
Done